src/type_translator.ts

```diff
@@ -245,12 +245,27 @@ export class TypeTranslator {
       this.ensureSymbolDeclared(sym);
     }
 
-    // This follows getSingleLineStringWriter in the TypeScript compiler.
+    const name = this.typeChecker.symbolToEntityName(
```

Comment:
The "other approaches" comments around L233 are probably worth dropping given that they've probably rotted, and the impl comment below here better describes what's going on anyway.

Reply:
Done.